datamodel/initdb.d/03_dcsa_im_v3_0.sql

```diff
@@ -448,7 +448,105 @@ CREATE TABLE dcsa_ebl_v1_0.shipment_transport (
 
 /* Events related Entities */
 
-
+DROP TABLE IF EXISTS dcsa_ebl_v1_0.event CASCADE;
+CREATE TABLE dcsa_ebl_v1_0.event (
+    event_id uuid DEFAULT uuid_generate_v4() PRIMARY KEY,
+    event_type text NOT NULL,
+    event_classifier_code varchar(3) NOT NULL,
+    event_date_time timestamp with time zone NOT NULL,
+    event_type_code varchar(4) NOT NULL
+);
+
+DROP TABLE IF EXISTS dcsa_ebl_v1_0.equipment_event CASCADE;
+CREATE TABLE dcsa_ebl_v1_0.equipment_event (
+    equipment_reference varchar(15),
+    empty_indicator_code text NOT NULL,
+    transport_call_id uuid NOT NULL
+) INHERITS (dcsa_ebl_v1_0.event);
+
+DROP TABLE IF EXISTS dcsa_ebl_v1_0.shipment_event CASCADE;
+CREATE TABLE dcsa_ebl_v1_0.shipment_event (
+    shipment_id uuid NOT NULL,
+    shipment_information_type_code varchar(3) NOT NULL
+) INHERITS (dcsa_ebl_v1_0.event);
+
+DROP TABLE IF EXISTS dcsa_ebl_v1_0.transport_event CASCADE;
+CREATE TABLE dcsa_ebl_v1_0.transport_event (
+    delay_reason_code varchar(3),
+    vessel_schedule_change_remark varchar(250),
+    transport_call_id uuid NOT NULL
+) INHERITS (dcsa_ebl_v1_0.event);
+
+DROP TABLE IF EXISTS dcsa_ebl_v1_0.event_subscription CASCADE;
+CREATE TABLE dcsa_ebl_v1_0.event_subscription (
+    id uuid DEFAULT uuid_generate_v4() PRIMARY KEY,
+    callback_url text NOT NULL,
+    event_type text, --This field must be able to contain multiple event types. Currently it does not.
+    booking_reference varchar(35),
+    transport_document_id varchar(20),
+    transport_document_type text,
+    equipment_reference varchar(15),
+    schedule_id uuid NULL,
+    transport_call_id uuid NULL
+    );
+
+
+-- Aggregated table containing all events
```

Comment:
This view is very implementation specific and as such part of the IM. Suggest it be moved to its own file.